transaction-spec/src/main/java/com/bloxbean/cardano/client/transaction/spec/Value.java

```diff
@@ -150,28 +154,149 @@ private Map serializeMultiAssets() {
      * @param that parameter to sum with
      * @return {@link Value} of the Sum
      */
-    public Value plus(Value that) {
+    public Value add(Value that) {
         BigInteger sumCoin = (getCoin() == null ? BigInteger.ZERO.add(that.getCoin()) : getCoin().add(that.getCoin()));
         List<MultiAsset> sumMultiAssets = MultiAsset.mergeMultiAssetLists(getMultiAssets(), that.getMultiAssets());
         return Value.builder().coin(sumCoin).multiAssets(sumMultiAssets).build();
     }
 
+    /**
+     * Use add(Value that) instead
+     *
+     * @param that
+     * @return
+     */
+    @Deprecated
+    public Value plus(Value that) {
+        return this.add(that);
+    }
+
+    public Value addLovelace(BigInteger amount) {
+        return this.add(fromLovelace(amount));
+    }
+
+    public Value add(String policyId, String assetName, BigInteger amount) {
+        return this.add(from(policyId, assetName, amount));
+    }
+
+    public Value add(String unit, BigInteger amount) {
+        if (unit == null || LOVELACE.equals(unit)) {
+            return add(fromLovelace(amount));
+        } else {
+            Tuple<String, String> policyAndAssetName = AssetUtil.getPolicyIdAndAssetName(unit);
+            return this.add(from(policyAndAssetName._1, policyAndAssetName._2, amount));
+        }
+    }
+
+
+    public static Value from(String policyId, String assetName, BigInteger amount) {
+        if (policyId == null || policyId.isBlank()) {
+            return fromLovelace(amount);
+        } else {
+            return Value.builder()
+                    .multiAssets(List.of(MultiAsset.builder()
+                            .policyId(policyId)
+                            .assets(List.of(Asset.builder().name(assetName).value(amount).build()))
+                            .build()))
+                    .build();
+        }
+    }
+
+    public static Value fromLovelace(BigInteger lovelaces) {
+        return Value.builder().coin(lovelaces).build();
+    }
+
     /**
      * Subtracts arbitrary complex values.
      *
      * @param that parameter to subtract by
      * @return {@link Value} Difference
      */
-    public Value minus(Value that) {
+    public Value subtract(Value that) {
         BigInteger sumCoin = (getCoin() == null ? BigInteger.ZERO.subtract(that.getCoin()) : getCoin().subtract(that.getCoin()));
         List<MultiAsset> difMultiAssets = MultiAsset.subtractMultiAssetLists(getMultiAssets(), that.getMultiAssets());
 
-        //Remove all asset with value == 0
-        difMultiAssets.forEach(multiAsset ->
-                multiAsset.getAssets().removeIf(asset -> BigInteger.ZERO.equals(asset.getValue())));
-        //Remove multiasset if there's no asset
-        difMultiAssets.removeIf(multiAsset -> multiAsset.getAssets() == null || multiAsset.getAssets().isEmpty());
+        List<MultiAsset> filteredMultiAssets = difMultiAssets
```

Comment:
This change is removing forEach and using stream api to avoid extra iteration. right?